test/test_pytree.py

```diff
@@ -163,24 +163,42 @@ def test_treemap(self):
         def run_test(pytree):
             def f(x):
                 return x * 3
-            sm1 = sum(map(tree_flatten(pytree)[0], f))
+            sm1 = list(map(f, tree_flatten(pytree)[0]))
             sm2 = tree_flatten(tree_map(f, pytree))[0]
             self.assertEqual(sm1, sm2)
 
             def invf(x):
                 return x // 3
+            sm3 = list(map(f, tree_flatten(pytree)[0]))
+            sm4 = tree_flatten(tree_map(f, pytree))[0]
+            self.assertEqual(sm3, sm4)
 
-            self.assertEqual(tree_flatten(tree_flatten(pytree, f), invf), pytree)
-
-            cases = [
```

Comment:
I think there were three errors in this test. So I modified the test case.
- The for loop was written inside the function, so these cases were never executed and run.
- In `sm1`, the map function takes 1st arg as function and the other as iter (the opposite was given earlier). Another thing here is if we sum(), it'll return the sum of all elements in lists, but we need to check if the returned lists are equal or not.
- At line 173, `tree_flatten` takes only one positional argument but here, two are given earlier.

 Please let me know if I need to be corrected. Thanks!

Reply:
Nice catch! You are right the tests were not run and hence we didn't catch the errors!

Thank you!